src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedData: GroupsMap<T> = {};
+  const allKeysValues: Array<string> = [];
+
+  for (const item of items) {
+    const value = `${item[key]}`;
+
+    if (!allKeysValues.includes(value)) {
+      allKeysValues.push(value);
+    }
+  }
+
+  for (const value of allKeysValues) {
+    groupedData[value] = items.filter((el) => `${el[key]}` === value);
```

Comment:
Using `Array.prototype.filter` inside a loop for each key is inefficient as it will iterate over the entire 'items' array for every key. Consider building the groups in a single iteration without using `filter`.